python/factor-selection-via-machine-learning.qmd

```diff
@@ -366,7 +366,7 @@ Cross-validation is a technique that allows us to alleviate this problem. We app
 How should you pick $K$? Large values of $K$ are preferable because the training data better imitates the original data. However, larger values of $K$ will have much higher computation time.
 `scikit-learn` provides all required tools to conduct $K$-fold cross-validation. We just have to update our model specification. In our case, we specify the penalty factor $\lambda$ as well as the mixing factor $\rho$ as *free* parameters.
 
-For our sample, we consider a time-series cross-validation sample. This means that we tune our models with 20 random samples of length five years with a validation period of four years. For a grid of possible hyperparameters, we then fit the model for each fold and evaluate $\hat{\text{MSPE}}$ in the corresponding validation set. Finally, we select the model specification with the lowest MSPE in the validation set. First, we define the cross-validation folds based on our training data only.
+For our sample, we consider a time-series cross-validation sample. This means that we tune our models with 20 samples of length five years with a validation period of four years. For a grid of possible hyperparameters, we then fit the model for each fold and evaluate $\hat{\text{MSPE}}$ in the corresponding validation set. Finally, we select the model specification with the lowest MSPE in the validation set. First, we define the cross-validation folds based on our training data only.
```

Comment:
I'm not sure if these are semantics, but in this specific case with time-series cross-validation, I actually do not see randomness. The cross-validation split determines which observation belongs to which fold, so I'd instead not refer to a random sample in this context. (Same for the Python version)

Reply:
The proposal is to remove the "random", so I interpret your statement as agreement :)

Reply:
Oh, never mind, I was too sloppy :-D 
Sure, agree 👍